src/Data/LLVM/BitCode/IR/Metadata.hs

```diff
@@ -534,15 +534,23 @@ parseMetadataEntry vt mt pm (fromEntry -> Just r) =
 
     13 -> label "METADATA_SUBRANGE" $ do
       assertRecordSizeIn [3, 5]
-      field0     <- parseField r 0 unsigned
-      let isDistinct = field0 /= 0
-      --  format = field0 `shiftR` 1
-      -- TODO: use the format to determine how to parse remaining fields.
-      -- The current parsing is likely to give the wrong results for bitcode
-      -- files generated by newer Clang versions, but it won't crash.
-      diNode         <- DISubrange
-        <$> parseField r 1 numeric     -- disrCount
-        <*> parseField r 2 signedInt64 -- disrLowerBound
+      field0 <- parseField r 0 unsigned
+      let isDistinct = field0 .&. 0 == 1
+      let format = field0 `shiftR` 1
+      ctx <- getContext
+      diNode <- case format of
+        2 -> do count <- mdForwardRefOrNull ctx mt <$> parseField r 1 numeric
```

Comment:
This `format` logic is tricky. It would be worth citing the relevant part of the LLVM codebase ([here](https://github.com/llvm/llvm-project/blob/d1c643a0168268445dab29d96b96d5fcb9c97f92/llvm/lib/Bitcode/Reader/MetadataLoader.cpp#L1437-L1445), I believe) from which this takes inspiration.

Reply:
Yup: added.